src/Stryker.Core/Stryker.Core.UnitTest/MutantFilters/IgnoredMethodMutantFilterTests.cs

```diff
@@ -14,13 +16,57 @@ namespace Stryker.Core.UnitTest.MutantFilters
 {
     public class IgnoredMethodMutantFilterTests : TestBase
     {
+
+
         [Fact]
         public static void ShouldHaveName()
         {
             var target = new IgnoredMethodMutantFilter() as IMutantFilter;
             target.DisplayName.ShouldBe("method filter");
         }
 
+        // filter should support documented cases
+        [Theory]
+        [InlineData("IgnoredMethod(true);", "true", true)]
+        [InlineData("x = IgnoredMethod(true);", null, true)]
+        [InlineData("var x = IgnoredMethod(true);", null, true)]
+        [InlineData("while (x == IgnoredMethod(true));", "==", false)]
+        [InlineData("IgnoredMethod()++;", "IgnoredMethod()++", false)]
+        [InlineData("x==1 ? IgnoredMethod(true): IgnoredMethod(false);", null, true)]
```

Comment:
What is this actually testing? Which mutation is ignored? The equality mutation or the boolean mutation?

Reply:
The equality mutation should not be ignored. The boolean mutations should be ignored.

Reply:
There is a type here. It should read: `[InlineData("x==1 ? IgnoredMethod(true): IgnoredMethod(false);", "==", true)]`.
I.e. the equality is mutated.
As said in the discussion: ternary expression mutation should be ignored if true and false expression (mutations) are to be ignored. If `IgnoredMethod` is somme logging method, this result in a unkillable mutant.
One could also note that 
`x!=1 ? IgnoredMethod(true): IgnoredMethod(false);` is semantically equivalent to `x==1 ? IgnoredMethod(false): IgnoredMethod(true);`. And the second one should be ignored because they alter the parameter. And yes, this is not a an actual mutation since I changed the parameters to both invocations, but I do not think this is important for the demonstration.

The underlying principle is that: if a method's mutation is to be ignored, one can assume that its (direct or side) effects cannot be tested, so all invocations are assumed to semantically equivalent.

May I suggest we continue discussing this in the discussion in #2669